### PR TITLE
Update PR template with task list for testing and documentation

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -2,5 +2,16 @@
 
 *Description of changes:*
 
+----
+*Testing*  
+- [ ] Did you run our unit tests? `make test`
```
Comment (pkandasamy91) [inline .github/PULL_REQUEST_TEMPLATE.md]: nit: our is relative, might be better to phrase it as the

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -2,5 +2,16 @@
...
+- [ ] Did you run our integration tests? `make integ-test`
+- [ ] Did you add new unit test(s)?
+- [ ] Did you add new integration test(s)?
```
Comment (PettitWesley) [inline .github/PULL_REQUEST_TEMPLATE.md]: How about we also add a section for listing manual test cases?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -2,5 +2,16 @@
 
 *Description of changes:*
 
+----
+*Testing*  
+- [ ] Did you run our unit tests? `make test`
+- [ ] Did you run our integration tests? `make integ-test`
```
Comment (allisaurus) [inline .github/PULL_REQUEST_TEMPLATE.md]: I would prefer to list checks as affirmative statements than questions:

* `make test` passes
* `make integ-test` passes
* Unit tests added for new functionality
* etc...

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (allisaurus) [reply .github/PULL_REQUEST_TEMPLATE.md]: Applicable cases could change depending on what functionality as added, but minimally we could have section like, "List manual checks done and their output (include default case, embellished case(s), new/old functionality, etc.)".

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -2,5 +2,16 @@
...
+- [ ] Did you add new unit test(s)?
+- [ ] Did you add new integration test(s)?
+----
+*Documentation*
```
Comment (allisaurus) [inline .github/PULL_REQUEST_TEMPLATE.md]: Add check for "Updated README if applicable"

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (efekarakus) [reply .github/PULL_REQUEST_TEMPLATE.md]: Added:

- [ ] Listed manual checks and their outputs in the comments ([example](https://github.com/aws/amazon-ecs-cli/pull/750#issuecomment-472623042))

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (efekarakus) [reply .github/PULL_REQUEST_TEMPLATE.md]: Sounds good this seems more Teutonic, I had taken my template from AppMesh. Updated.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (efekarakus) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

**Pull request merged**